sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/SqlKeywordsDelimitedBySpace.cs

```diff
@@ -157,41 +155,58 @@ public override void VisitBinaryExpression(BinaryExpressionSyntax node)
                 }
             }
 
-            private void CheckSpaceBetweenStrings(List<LiteralExpressionSyntax> stringLiterals)
+            private void CheckSpaceBetweenStrings(List<StringWrapper> stringWrappers)
             {
-                for (var i = 0; i < stringLiterals.Count -1; i++)
+                for (var i = 0; i < stringWrappers.Count -1; i++)
                 {
-                    var firstStringText = stringLiterals[i].Token.ValueText;
-                    var secondString = stringLiterals[i + 1];
-                    var secondStringText = secondString.Token.ValueText;
+                    var firstStringText = stringWrappers[i].Text;
+                    var secondString = stringWrappers[i + 1];
+                    var secondStringText = secondString.Text;
                     if (firstStringText.Length > 0 &&
                         IsAlphaNumericOrAt(firstStringText.ToCharArray().Last()) &&
                         secondStringText.Length > 0 &&
                         IsAlphaNumericOrAt(secondStringText[0]))
                     {
                         var word = secondStringText.Split(' ').FirstOrDefault();
-                        this.context.ReportDiagnosticWhenActive(Diagnostic.Create(rule, secondString.GetLocation(), word));
+                        this.context.ReportDiagnosticWhenActive(Diagnostic.Create(rule, secondString.Node.GetLocation(), word));
                     }
                 }
             }
 
+            private static bool TryGetStringWrapper(ExpressionSyntax expression, out StringWrapper stringWrapper)
+            {
+                if (expression is LiteralExpressionSyntax literal && literal.IsKind(SyntaxKind.StringLiteralExpression))
+                {
+                    stringWrapper = new StringWrapper(literal, literal.Token.ValueText);
+                    return true;
+                }
+                else if (expression is InterpolatedStringExpressionSyntax interpolatedString)
```

Comment:
I think we can remove the else here since we do a direct return in the first if.

An option would be to use a switch here since we already have three branches but this is just a mater of preferences.